job-service-db/src/main/resources/db/migration/procedures/R___old.sql

```diff
@@ -0,0 +1,276 @@
+--
```

Comment:
As per [Flyway documentation](https://flywaydb.org/documentation/concepts/migrations.html#repeatable-migrations): Within a single migration run, repeatable migrations are always applied last, after all pending versioned migrations have been executed. Repeatable migrations are applied in the order of their description.

Reply:
Quick note for that [article](https://flywaydb.org/blog/flyway-timestampsandrepeatables) which could bring an element of response

Reply:
A repeatable migration in Flyway is a migration that is run whenever its checksum changes. The scripts run in alphabetical order if their checksum is changed. This is similar to what is currently happening in our Liquibase configuration. Only that in Liquibase, we would solve some order issues by grouping a function depending on another into a same changeset so we control that order. We might want to change the filenaming convention in the future, but I could see only a single situation where we have to do that in Liquibase. [Here](https://github.com/JobService/job-service/blob/6817f9dfa8fea2df623a99546a533c6b73c2cca5/job-service-db/src/main/resources/procedures/changelog.xml#L78) 
About the introduction of a ___old.sql file where we could set the obsolete functions and signatures, I believe that it could be a good improvement for readability in the different scripts.
However, it could be error prone as we would have to do with 2 different files when changing a signature.
@dermot-hardy what do you think? I reversed to match the current configuration in order to minimize the changes, but kept the modification in a separate branch if we want to put them in.

Reply:
> If I understand correctly, we are at present depending on the filenames to assure the order in which the scripts are executed.

We shouldn't be dependent on this.  If we're going to use repeatable migrations then they should be authored so that they can run in any order.  (Otherwise we may as well be using versioned migrations.)

Reply:
I don't know if there's a cleaner way but I think the way I would handle this is to include forward declarations in the modules that use other functions.  The example you gave above is that `internal_does_table_exist` makes use of `internal_to_regclass` internally.  I would declare (not define) the `internal_to_regclass` function in the `R__doesTableExist.sql` file, before the definition of the `internal_does_table_exist` function - i.e. using something like this:

```
DO $$
BEGIN
    CREATE FUNCTION internal_to_regclass(rel_name VARCHAR(63))
    RETURNS regclass
    LANGUAGE plpgsql STABLE
    AS 'BEGIN /* Forward Declaration */ END';
EXCEPTION WHEN duplicate_function THEN
END$$;
```

That way the order that the repeatable migrations are run in should not matter, and if either implementation changes in the future it should not matter either.